src/jit/lsraxarch.cpp

```diff
@@ -2297,224 +2297,246 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
-            assert(dstCount == 1);
-            tgtPrefUse = BuildUse(op1);
-            buildUses  = false;
-            break;
+            if (Compiler::isImmHWIntrinsic(intrinsicID, lastOp) && !lastOp->isContainedIntOrIImmed())
+            {
+                assert(!lastOp->IsCnsIntOrI());
+
+                // We need two extra reg when lastOp isn't a constant so
+                // the offset into the jump table for the fallback path
+                // can be computed.
+                buildInternalIntRegisterDefForNode(intrinsicTree);
+                buildInternalIntRegisterDefForNode(intrinsicTree);
+            }
         }
 
-        case NI_AVX_SetAllVector256:
+        // Determine whether this is an RMW operation where op2+ must be marked delayFree so that it
+        // is not allocated the same register as the target.
+        bool isRMW = intrinsicTree->isRMWHWIntrinsic(compiler);
+
+        // Create internal temps, and handle any other special requirements.
+        // Note that the default case for building uses will handle the RMW flag, but if the uses
+        // are built in the individual cases, buildUses is set to false, and any RMW handling (delayFree)
+        // must be handled within the case.
+        switch (intrinsicID)
         {
-            if (varTypeIsIntegral(baseType))
+            case NI_SSE_CompareEqualOrderedScalar:
+            case NI_SSE_CompareEqualUnorderedScalar:
+            case NI_SSE_CompareNotEqualOrderedScalar:
+            case NI_SSE_CompareNotEqualUnorderedScalar:
+            case NI_SSE2_CompareEqualOrderedScalar:
+            case NI_SSE2_CompareEqualUnorderedScalar:
+            case NI_SSE2_CompareNotEqualOrderedScalar:
+            case NI_SSE2_CompareNotEqualUnorderedScalar:
             {
-                buildInternalFloatRegisterDefForNode(intrinsicTree, allSIMDRegs());
-                if (!compiler->compSupports(InstructionSet_AVX2) && varTypeIsByte(baseType))
+                buildInternalIntRegisterDefForNode(intrinsicTree, allByteRegs());
+                setInternalRegsDelayFree = true;
+                break;
+            }
+
+            case NI_SSE_SetScalarVector128:
+            case NI_SSE2_SetScalarVector128:
+            {
+                buildInternalFloatRegisterDefForNode(intrinsicTree);
+                setInternalRegsDelayFree = true;
+                break;
+            }
+
+            case NI_SSE_ConvertToSingle:
+            case NI_SSE_StaticCast:
+            case NI_SSE2_ConvertToDouble:
+            case NI_AVX_ExtendToVector256:
+            case NI_AVX_GetLowerHalf:
+            case NI_AVX_StaticCast:
+            {
+                assert(dstCount == 1);
+
+                if (!op1->isContained())
                 {
-                    buildInternalFloatRegisterDefForNode(intrinsicTree, allSIMDRegs());
+                    tgtPrefUse = BuildUse(op1);
+                    srcCount   = 1;
                 }
+
+                buildUses = false;
```

Comment:
This will cause uses not to be built in the case where `op1` is a contained memory op. I'm pretty sure that case needs to be handled, so I suspect that the `buildUses = false` belongs in the if-clause.